src/woodbury_pd_mat.jl

```diff
@@ -88,3 +88,7 @@ end
 # NOTE: the parameterisation to scale up the Woodbury matrix is not unique. Here we
 # implement one way to scale it.
 *(a::WoodburyPDMat, c::Real) = WoodburyPDMat(a.A, a.D * c, a.S * c)
+*(c::Real, a::WoodburyPDMat) = a * c
+*(a::WoodburyPDMat, c::Diagonal{T}) where {T<:Real} = WoodburyPDMat(sqrt(c) * a.A, a.D, a.S * c)
```

Comment:
what about `c` has negative values?

Also, as `*` calculation is not unique to Woodbury, it's probably better to have some internal consistency for the implementation here. Now we adjust `D` for Real *, and `A` for Diagonal *, which is a bit weird

Reply:
Unless I'm mistaken, if `c` is negative that means that the resultant cannot be a PDMat. This kind of error handling needs to be handling at the constructor. So it's fine to assume `c` has all positive elements. 

Regarding `a.D * c`, these are not comformable. The size of D is `R ^ R`, the size of `c` is `N^N`. `A` is `N ^ R`, hence doing `c * A`: ( ( N ^ N) * (N ^ R) ) -> ( N ^ R))

Reply:
fair point, but if `c` has negative values, you'll hit error when calling `sqrt(c)`, before you construct WoodburyPDMat. BTW, the current implementation of WoodburyPDMat doesn't validate the values of `A`, so the issue you raised won't be captured in the constructor

Reply:
> c has negative values, you'll hit error when calling sqrt(c),

🤷 Allright, i've updated this so it checks whether the diagonals are posdef.